packages/nextjs/src/server/utils/runClerkMiddleware.ts

```diff
@@ -0,0 +1,141 @@
+import type { Base } from '@clerk/backend-core';
+import { AuthStatus, createGetToken } from '@clerk/backend-core';
+import { NextURL } from 'next/dist/server/web/next-url';
+import { NextRequest, NextResponse } from 'next/server';
+
+import { AuthData } from '../../middleware/types';
+import { AuthStatusMapping, SessionsApi } from '../types';
+
+const DEFAULT_API_URL = process.env.CLERK_API_URL || 'https://api.clerk.dev';
+const DEFAULT_API_VERSION = process.env.CLERK_API_VERSION || 'v1';
+
+const INTERSTITIAL_URL = `${DEFAULT_API_URL}/${DEFAULT_API_VERSION}/internal/interstitial`;
+
+type RunClerkMiddlewareOptions = {
+  jwtKey?: string;
+  authorizedParties?: string[];
+};
+
+// TODO Extend NextResponse instead?
+export type ClerkNextResponse = {
+  next: () => NextResponse;
+  error: () => NextResponse;
+  rewrite: () => NextResponse;
+  redirect: () => NextResponse;
+};
+
+// TODO middleware will need to discern error (interstitial case) from signed out case
+export type RunClerkMiddlewareReturn = {
+  auth?: AuthData; // TODO need a more limited type without user & organization, but possibly with isSignedIn?
+  response?: ClerkNextResponse;
+};
+
+export function createRunClerkMiddleware(base: Base, sessionsApi: SessionsApi) {
```

Comment:
🙃 Consider using an object here as more options might come soon

Reply:
Not applicable any more after latest changes